### PR TITLE
Import DLC with uniquebodyparts, add Tracks

### PR DESCRIPTION
**Pull request by getzze**
### Description
Importing Deeplabcut project with multiple animals failed because:
- the video has a ".avi" extension and current code assumes .mp4 extenstion.
- the DLC config file has a non-empty `uniquebodyparts` list. Unique body parts do not belong to any individual, so parsing the .csv file containing the labeled frames leads to key error of the type: Cannot find key ("animal", "unique_body_part"). Unique body parts belong to the "single" individual.

In addition, this PR creates Tracks with the animal names (and "single" for the unique body parts), which is a new feature.

I added tests for importing Multiple animal DLC with uniquebodyparts. 

### Types of changes

- [X] Bugfix
- [X] New feature
- [ ] Refactor / Code style update (no logical changes)
- [ ] Build / CI changes
- [ ] Documentation Update
- [ ] Other (explain)

### Does this address any currently open issues?
#734 

### Outside contributors checklist

- [ ] Review the [guidelines for contributing](https://github.com/talmolab/sleap/blob/develop/docs/CONTRIBUTING.md) to this repository
- [ ] Read and sign the [CLA](https://github.com/talmolab/sleap/blob/develop/sleap-cla.pdf) and add yourself to the [authors list](https://github.com/talmolab/sleap/blob/develop/AUTHORS)
- [ ] Make sure you are making a pull request against the **develop** branch (not *main*). Also you should start *your branch* off *develop*
- [ ] Add tests that prove your fix is effective or that your feature works
- [ ] Add necessary documentation (if appropriate)

#### Thank you for contributing to SLEAP!
:heart:

### Summary by CodeRabbit

- New Feature: Enhanced the software's ability to handle different video file extensions, improving the flexibility of video file input.
- Improvement: Updated the system to store individual animal tracks in a dictionary, enhancing the organization and accessibility of data.
- Bug Fix: Corrected a spelling error in the `summary()` function, improving the readability of the code.
- Test: Added a new test case `test_maudlc` to ensure the correct parsing and organization of data from CSV files.
- Test: Updated existing test cases to align with new data structures and functionalities, ensuring the reliability of the software.
- Refactor: Replaced the `animal_names` list with a `tracks` dictionary for better data management.

### CODE REVIEW

File: tests/io/test_formats.py [hunk 1]
```diff
@@ -232,6 +232,77 @@
...
+        [[np.nan, np.nan], [np.nan, np.nan], [np.nan, np.nan], [22, 23], [24, 25]],
+    )
+    assert_array_equal(
+        labels[2][0].numpy(),
+        [[26, 27], [28, 29], [30, 31], [np.nan, np.nan], [np.nan, np.nan]],
+    )
+    assert_array_equal(
+        labels[2][0].numpy(),
+        [[np.nan, np.nan], [np.nan, np.nan], [np.nan, np.nan], [32, 33], [34, 35]],
+    )
```
[the remark this answers is not in the document]
Comment (getzze) [reply tests/io/test_formats.py]: good catch!

File: tests/io/test_formats.py [hunk 2]
```diff
@@ -232,6 +232,77 @@
...
+    assert labels[2].frame_idx == 3
+
+    # Assert tracks are correct
+    assert len(labels.tracks) == 3
+    assert set(t.name for t in labels.tracks) == set(["Animal1", "Animal1", "single"])
+    for t in labels.tracks:
+        if t.name == "single":
+            assert t.spawned_on == 1
+        else:
+            assert t.spawned_on == 0
```
[the remark this answers is not in the document]
Comment (getzze) [reply tests/io/test_formats.py]: another copy-paste

**Comment by talmo** [comment]
Thanks for the contribution @getzze! Looks like there's still a failing test related to the tracks, but just ping us when you're ready for review :)

**Comment by getzze** [comment]
@talmo all the tests pass now, it's ready for reviewing.

**Review by talmo — APPROVED** [review]
Looks great to me!

**Comment by MHRosenberg** [comment]
Hi there, I'm attempting to train a fresh SLEAP (single animal) model from my existing DLC labeled frames. The import appears to have worked beautifully! Nice job! However, I currently get the issue below, presumably because SLEAP is not expecting variations in the frame dimensions for the input images. I'm going to try the following hacks in the following order:

1. resize them all to the same dimensions
2. cropping all of the images to be same dimensions
3. sorting them by frame dimensions. 
I'll post my experiences in future comments unless someone already has a fix they can refer me to. 

It'd be awesome if SLEAP could anticipate this issue and handle it gracefully. 

Traceback (most recent call last):
  File "/Users/mhr3/miniconda3/envs/sleap/lib/python3.9/site-packages/sleap/gui/learning/dialog.py", line 810, in export_package
    ret = sleap.gui.commands.export_dataset_gui(
  File "/Users/mhr3/miniconda3/envs/sleap/lib/python3.9/site-packages/sleap/gui/commands.py", line 1425, in export_dataset_gui
    Labels.save_file(
  File "/Users/mhr3/miniconda3/envs/sleap/lib/python3.9/site-packages/sleap/io/dataset.py", line 1997, in save_file
    write(filename, labels, *args, **kwargs)
  File "/Users/mhr3/miniconda3/envs/sleap/lib/python3.9/site-packages/sleap/io/format/main.py", line 162, in write
    return disp.write(filename, source_object, *args, **kwargs)
  File "/Users/mhr3/miniconda3/envs/sleap/lib/python3.9/site-packages/sleap/io/format/dispatch.py", line 79, in write
    return adaptor.write(filename, source_object, *args, **kwargs)
  File "/Users/mhr3/miniconda3/envs/sleap/lib/python3.9/site-packages/sleap/io/format/hdf5.py", line 247, in write
    new_videos = labels.save_frame_data_hdf5(
  File "/Users/mhr3/miniconda3/envs/sleap/lib/python3.9/site-packages/sleap/io/dataset.py", line 2369, in save_frame_data_hdf5
    vid = video.to_hdf5(
  File "/Users/mhr3/miniconda3/envs/sleap/lib/python3.9/site-packages/sleap/io/video.py", line 1433, in to_hdf5
    frame_data = self.get_frames(frame_numbers)
  File "/Users/mhr3/miniconda3/envs/sleap/lib/python3.9/site-packages/sleap/io/video.py", line 1117, in get_frames
    return np.stack([self.get_frame(idx) for idx in idxs], axis=0)
  File "<__array_function__ internals>", line 180, in stack
  File "/Users/mhr3/miniconda3/envs/sleap/lib/python3.9/site-packages/numpy/core/shape_base.py", line 426, in stack
    raise ValueError('all input arrays must have the same shape')
ValueError: all input arrays must have the same shape

**Comment by talmo** [comment]
Hi @MHRosenberg,

This is a pull request related to implementing aspects of the DLC import functionality, so not quite the right place for your bug report, but I did create an issue over here that you can follow: https://github.com/talmolab/sleap/issues/2085

I'll continue the discussion over there. Thanks for the report!

Cheers,

Talmo

**Pull request merged**